ouroboros-network/src/Ouroboros/Network/Diffusion.hs

```diff
@@ -6,122 +6,152 @@
 module Ouroboros.Network.Diffusion
   ( -- * Common API
     P2P (..)
-  , DiffusionTracer (..)
-  , Tracers (..)
-  , nullTracers
   , ExtraTracers (..)
-  , Failure (..)
-  , Arguments (..)
-  , ExtraArguments (..)
+  , ArgumentsExtra (..)
   , Applications (..)
-  , ExtraApplications (..)
+  , ApplicationsExtra (..)
     -- * Run data diffusion
   , run
-    -- * Re-exports
-  , P2P.AbstractTransitionTrace
-  , PublicPeerSelectionState
-  , makePublicPeerSelectionStateVar
   ) where
 
-import Control.Exception (IOException)
-import Data.Functor (void)
-
-import Network.Socket (Socket)
-
-import Ouroboros.Network.NodeToClient (LocalAddress, LocalSocket,
-           NodeToClientVersion, NodeToClientVersionData)
-import Ouroboros.Network.NodeToNode (NodeToNodeVersion, NodeToNodeVersionData,
-           RemoteAddress)
-import Ouroboros.Network.PeerSelection.Governor.Types
-
+import Cardano.Diffusion.P2P qualified as P2P
```

Comment:
It should be the other way around.  Here (or rather in `Ouroboros.Network.Diffusion.P2P`) we should expose the most general diffusion function, which is used by `Cardano.Diffusion.P2P` with cardano specific instantiations & callbacks: this might require adding some extra callbacks needed in the cardano case (they might be useful in other cases too).